src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -27,7 +27,10 @@ public ConsoleLoggerProcessor()
 
         public virtual void EnqueueMessage(LogMessageEntry message)
         {
-            _messageQueue.Add(message);
+            if (!_messageQueue.IsAddingCompleted)
+            {
+                _messageQueue.Add(message);
```

Comment:
I'm not sure how often this happens in production or if it ever does, but `Add` blocks if the queue is full, so this could throw still.